leetcode/medium/227._Basic_Calculator_II.py

```diff
@@ -0,0 +1,51 @@
+# Link to problem: https://leetcode.com/problems/basic-calculator-ii/description/
+# Loom explaining solution you wrote (5m): 
+# Solution:
+# Description
+# Key ideas
+# To step process, process */ first, +- last.
+# - can be treated as + (-val), +- can be mapped as sum
+# When processing */, use a value stack
+# Difficulty: Medium
+# Code
+
+
+def token_of(s):
+    cur = ""
+    for c in s:
+        if c == " ":
```

Comment:
This check adds additional nesting to your if/else statements. 

What you're doing is basically removing all of the space characters. You can do this in the beginning by using `.replace()` or `.split()`

Reply:
You are right, to me scanning characters is more intuitive, but both works.

Reply:
Using `.replace()` or `.split()` in this case is much better than scanning individual characters. 

You remove the risk of error and it makes it more clear what your code is doing.

Please make this change